File: front/apps/web-antd/src/api/userecho/customer.ts
/**
 * 客户管理 API
 */

import { requestClient } from '#/api/request';

/** 客户对象 */
export interface Customer {
  id: string;
  tenant_id: string;
  name: string;
  customer_type: string;
  business_value: number;
  created_time: string;
  updated_time: string;
  deleted_at?: string;
}

/** 创建客户参数 */
export interface CreateCustomerParams {
  name: string;
  customer_type?: string;
  business_value?: number;
}

/** 更新客户参数 */
export interface UpdateCustomerParams {
  name?: string;
  customer_type?: string;
  business_value?: number;
}

/** 客户列表查询参数 */
export interface CustomerListParams {
  skip?: number;
  limit?: number;
  search?: string;
  customer_type?: string;
}

/** 客户列表响应 */
export interface CustomerListResponse {
  items: Customer[];
  total: number;
}

/**
 * 获取客户列表
 */
export async function getCustomerList(params: CustomerListParams) {
  return requestClient.get<CustomerListResponse>('/api/v1/app/customers', { params });
}

/**
 * 搜索客户（模糊匹配）
 */
export async function searchCustomers(query: string, limit: number = 10) {
  return requestClient.get<Customer[]>('/api/v1/app/customers/search', { params: { query, limit } });
}

/**
 * 创建客户
 */
export async function createCustomer(data: CreateCustomerParams) {
  return requestClient.post<Customer>('/api/v1/app/customers', data);
}

/**
 * 更新客户
 */
export async function updateCustomer(id: string, data: UpdateCustomerParams) {
  return requestClient.put<Customer>(`/api/v1/app/customers/${id}`, data);
}

/**
 * 删除客户
 */
export async function deleteCustomer(id: string) {
  return requestClient.delete(`/api/v1/app/customers/${id}`);
}

/** 客户类型选项 */
export const CUSTOMER_TYPES = [
  { value: 'normal', label: '普通客户', business_value: 1, color: '#8c8c8c' },
  { value: 'paid', label: '付费客户', business_value: 3, color: '#1677ff' },
  { value: 'major', label: '大客户', business_value: 5, color: '#faad14' },
  { value: 'strategic', label: '战略客户', business_value: 10, color: '#722ed1' },
];

/** 商业价值等级配置 */
export const BUSINESS_VALUE_LEVELS = [
  { min: 1, max: 2, label: '低', color: '#8c8c8c', bgColor: 'rgba(140, 140, 140, 0.1)' },
  { min: 3, max: 4, label: '中', color: '#1677ff', bgColor: 'rgba(22, 119, 255, 0.1)' },
  { min: 5, max: 7, label: '高', color: '#faad14', bgColor: 'rgba(250, 173, 20, 0.1)' },
  { min: 8, max: 10, label: '核心', color: '#722ed1', bgColor: 'rgba(114, 46, 209, 0.15)' },
];

File: server/backend/app/userecho/schema/customer.py
from datetime import datetime

from pydantic import Field

from backend.common.schema import SchemaBase


class CustomerBase(SchemaBase):
    """客户基础模型"""

    name: str = Field(description="客户名称")
    customer_type: str = Field(
        default="normal", description="客户类型: normal=普通, paid=付费, major=大客户, strategic=战略客户"
    )
    business_value: int = Field(default=1, ge=1, le=10, description="商业价值权重 (1-10)")


class CustomerCreate(CustomerBase):
    """创建客户参数"""


class CustomerUpdate(SchemaBase):
    """更新客户参数"""

    name: str | None = Field(None, description="客户名称")
    customer_type: str | None = Field(None, description="客户类型")
    business_value: int | None = Field(None, ge=1, le=10, description="商业价值权重")


class CustomerOut(CustomerBase):
    """客户输出模型"""

    model_config = {"from_attributes": True}

    id: str = Field(description="客户ID")
    tenant_id: str = Field(description="租户ID")
    created_time: datetime = Field(description="创建时间")
    updated_time: datetime | None = Field(None, description="更新时间")
    deleted_at: datetime | None = Field(None, description="删除时间")
